public/src/components/ManageSubmissionStatus.tsx

```diff
@@ -18,25 +19,48 @@ const ManageSubmissionStatus = (): JSX.Element => {
 
 
     const handleRebuild = async () => {
+        if (rebuilding) { return } // Don't allow multiple rebuilds at once
         setRebuilding(true)
         await actions.rebuildSubmission()
         setRebuilding(false)
     }
 
+    const handleSetStatus = async (status: Submission_Status) => {
+        if (updating !== Submission_Status.NONE) { return } // Don't allow multiple updates at once
+        setUpdating(status)
+        await actions.updateSubmission({ owner: state.submissionOwner, submission: state.selectedSubmission, status })
+        setUpdating(Submission_Status.NONE)
+    }
 
     if (assignment && !isManuallyGraded(assignment)) {
-        buttons.push({ text: rebuilding ? "Rebuilding..." : "Rebuild", status: Submission_Status.NONE, style: rebuilding ? "secondary" : "primary", onClick: handleRebuild })
+        // Add rebuild button if the assignment is not manually graded
+        buttons.push({ text: rebuilding ? "Rebuilding..." : "Rebuild", status: -1, style: rebuilding ? "secondary" : "primary", onClick: handleRebuild })
```

Comment:
To make this more readable, put `text`, `status`, `style` and `onClick` on separate lines. As is, it is hard to parse where the `:` and `?` operators belong and so forth.

Also, why is `status: -1` instead of `NONE`?